crates/l2/contracts/src/l1/CommonBridge.sol

```diff
@@ -214,64 +205,28 @@ contract CommonBridge is
 
     /// @inheritdoc ICommonBridge
     function deposit(
-        uint256 _amount,
         address l2Recipient
     ) public payable override whenNotPaused {
-        uint256 value;
-
-        // Here we define value depending on whether the native token is ETH or an ERC20
-        if (NATIVE_TOKEN_L1 == address(0)) {
-            require(
-                msg.value > 0,
-                "CommonBridge: the native token is ETH, msg.value must be greater than zero"
-            );
-
-            require(
-                _amount == 0,
-                "CommonBridge: the native token is ETH, _amount must be zero"
-            );
-
-            value = msg.value;
-        } else {
-            require(
-                msg.value == 0,
-                "CommonBridge: the native token is an ERC20, msg.value must be zero"
-            );
-
-            require(
-                _amount > 0,
-                "CommonBridge: the native token is an ERC20, _amount must be greater than zero"
-            );
-
-            value = _amount;
-
-            // We lock the tokens in the bridge contract
-            IERC20(NATIVE_TOKEN_L1).transferFrom(
-                msg.sender,
-                address(this),
-                value
-            );
-        }
-
-        deposits[NATIVE_TOKEN_L2][NATIVE_TOKEN_L2] += value;
+        _deposit(l2Recipient);
+    }
 
+    function _deposit(address l2Recipient) private {
```

Comment:
We can now have `_deposit` body directly in `deposit` and call `deposit()` on `receive()`